stories/ranking.stories.jsx

```diff
@@ -0,0 +1,189 @@
+    await step('"Least" option not selectable; does not set item value', async () => {
+      expect(item.dataset.value).toBe(defaultValue)
+    })
+  })
+}
+
+const testEnabledWithDefaultNeutral = async (step, numericalIdentifier = '', defaultValue = 'Neutral') => {
+  let item
+  await common.asyncSleep(600)
+
+  await step('Item should be found', async () => {
+    item = document.querySelector(`#testRanking-${numericalIdentifier}`)
+    await step('Item DOM Node should be found', async () => {
+      expect(item).toBeTruthy()
+    })
+    await step(`Initial value should be ${defaultValue}`, async () => {
+      expect(item.dataset.value).toBe(defaultValue)
+    })
+  })
+
+  await step('When not disabled, states should be properly selectable and represented in item value', async () => {
+    await userEvent.click(item.querySelector('#rankingMost'))
+    await step('"Most" option selectable and sets item value', async () => {
+      expect(item.dataset.value).toBe('Most')
+    })
+
+    await userEvent.click(item.querySelector('#rankingNeutral'))
+    await step('"Neutral" option selectable and sets item value', async () => {
+      expect(item.dataset.value).toBe('Neutral')
+    })
+
+    await userEvent.click(item.querySelector('#rankingLeast'))
+    await step('"Least" option selectable and sets item value', async () => {
+      expect(item.dataset.value).toBe('Least')
+    })
+  })
+}
+
+const testDisabledWithDefaultNeutral = async (step, numericalIdentifier = '', defaultValue = 'Neutral') => {
+  let item
+  await common.asyncSleep(600)
+
+  await step('Item should be found', async () => {
+    item = document.querySelector(`#testRanking-${numericalIdentifier}`)
+    await step('Item DOM Node should be found', async () => {
+      expect(item).toBeTruthy()
+    })
+    await step(`Initial value should be ${defaultValue}`, async () => {
+      expect(item.dataset.value).toBe(defaultValue)
+    })
+  })
+
+  await step('When disabled, states should not be selectable and item value should not change', async () => {
+    await userEvent.click(item.querySelector('#rankingMost'))
+    await step('"Most" option not selectable; does not set item value', async () => {
+      expect(item.dataset.value).toBe(defaultValue)
+    })
+
+    await userEvent.click(item.querySelector('#rankingNeutral'))
+    await step('"Neutral" option not selectable; does not set item value', async () => {
+      expect(item.dataset.value).toBe(defaultValue)
+    })
+
+    await userEvent.click(item.querySelector('#rankingLeast'))
+    await step('"Least" option not selectable; does not set item value', async () => {
+      expect(item.dataset.value).toBe(defaultValue)
+    })
+  })
+}
+
+const generateComponent = (numericalIdentifier, disabled = false, defaultValue = '') => {
+  return (
+    <Ranking
+      disabled={disabled}
+      defaultValue={defaultValue}
+      id={`testRanking-${numericalIdentifier}`}
```

Comment:
I'm hoping we can have the label, but not require a unique id for every component.